packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.yaml

```diff
@@ -23,6 +23,15 @@ params:
     type: object
     required: false
     description: HTML attributes (for example data attributes) to add to the exit this page container.
+  - name: button
+    type: object
+    required: false
+    description: Options for the exit this page button.
+    params:
+      - name: attributes
+        type: object
+        required: false
+        description: HTML attributes (for example data attributes) to add to the exit this page button. The default disables referrer information being provided to the `redirectUrl` and tells search engines to ignore the page being linked to.
```

Comment:
I feel like we should use `isComponent` but it becomes a worse example of [**Radio and checkbox item nested label components list options that can't be used**](https://trello.com/c/RxuFrKQS/19-radio-and-checkbox-item-nested-label-components-list-options-that-cant-be-used) 🤦‍♂️

```suggestion
        description: HTML attributes (for example data attributes) to add to the exit this page button. The default disables referrer information being provided to the `redirectUrl` and tells search engines to ignore the page being linked to.
        isComponent: true
```

Reply:
That issue is pretty much why I didn't use it. We're only allowing customisation of a single one of the component's thirteen parameters, so linking off to the full component docs feels iffy.

I'm interested in your opinion on the feasibility of a merge function or filter, as I've mentioned above. That would create the possibility of a user customising every parameter without letting them override the bits required for the component to function. 

For a contrived example:

```nunjucks
{# Our preferred parameter values, but ones the user can override if they wish. #}
{% set defaultParams = {
  redirectUrl: 'https://www.bbc.co.uk/weather',
  button: {
    classes: 'govuk-button--warning',
    attributes: { 'rel': 'nofollow noreferrer' }
  }
} %}

{# The parameter values required for the component to work properly. #}
{% set requiredParams = {
  button: {
    classes: 'govuk-exit-this-page__button govuk-js-exit-this-page-button'
  }
} %}

{{ govukExitThisPage(mergeParams(
  defaultParams,
  params,
  requiredParams
)) }}
```

There's definitely some quirks there that need thought (like how user-defined classes would override `defaultParams` classes but be merged with `requiredParams` classes), but it would seem to make situations like this a bit tidier, perhaps!

Reply:
I'd love a merge function, but yeah probably a filter

We've had [little mentions of merging](https://github.com/alphagov/govuk-frontend/issues/995) for a while [on different issues](https://github.com/alphagov/govuk-frontend/pull/1172#issuecomment-460257465)

Every service team seems to maintain their own to merge POST form data etc:

```js
{{ govukInput(form.context.email | govukParams({
  type: "email",
  autocomplete: "username",
  spellcheck: false
})) }}
```

Reply:
I'm quite adamant that we shouldn't use `isComponent` here. A huge part of ETP's design success is specifically how we use the button component ie: it's a link that looks like a button and it's red. Allowing users to change that introduces risk. Putting my foot down on this suggestion not getting included.

Reply:
Foot down, agreed, suggestion wasn't to enable button options.

Rather, we have the `isComponent` flag to enable the "this is a component option" UI

<br>
<img width="760" alt="Button nested option" src="https://github.com/alphagov/govuk-frontend/assets/415517/e7f511bf-8d9e-4985-a3bd-1497831e6ddd">

Reply:
I've opted to hardcode the attribute for now, so this discussion is moot — or one for later, at least.

Reply:
Yeah definitely, would you mind expanding the [Trello card](https://trello.com/c/RxuFrKQS/19-radio-and-checkbox-item-nested-label-components-list-options-that-cant-be-used) to include Exit this Page buttons too?

Or adding a new one for nested components that we choose not to identify

Reply:
Aha gotcha. I'm on the fence about this as it's not really the component API we're exposing, just the attributes bit.

As it happens, it looks like this might not be relevant anymore from the above chat.

Reply:
Yeah no worries. What we really need is a way to document:

1. The component has a nested component option
2. The nested component option can configure all/some/one params
3. The nested component links to its page or shows options in a table (like `label` and `hint`)

But not for this PR

Reply:
I agree. I am far from against finding a way to communicate "we are using one of our components here and are letting you manipulate XYZ part of it".